src/inmanta/data/dataview.py

```diff
@@ -470,6 +471,27 @@ def __init__(
         self.environment = env
         self.deploy_summary = deploy_summary
 
+        # Rewrite the filter to special case orphan handling
+        # This doesn't affect the paging links, as they use the raw filter
+
+        status_filter_type, status_filter_fields = self.filter.get("status", (None, {}))
+        assert status_filter_type is None or status_filter_type == inmanta.data.QueryType.COMBINED
+        assert isinstance(status_filter_fields, dict)
+        self.drop_orphans = "orphaned" in status_filter_fields.get(inmanta.data.QueryType.NOT_CONTAINS, []) or (
+            "orphaned" not in status_filter_fields.get(inmanta.data.QueryType.CONTAINS, ["orphaned"])
+        )
+
+        if self.drop_orphans:
+            # clean filter for orphans
+            try:
+                status_filter_fields.get(inmanta.data.QueryType.NOT_CONTAINS, []).remove("orphaned")
+                if not status_filter_fields.get(inmanta.data.QueryType.NOT_CONTAINS, []):
+                    del status_filter_fields[inmanta.data.QueryType.NOT_CONTAINS]
+                if not status_filter_fields:
+                    del self.filter["status"]
+            except ValueError:
+                pass
```

Comment:
Out of curiosity, do you happen to know if it made a difference to keep this?

Reply:
Nothing I could measure